frame/support/procedural/src/construct_runtime/expand/origin.rs

```diff
@@ -140,7 +141,9 @@ pub fn expand_outer_origin(
 			}
 
 			fn filter_call(&self, call: &Self::Call) -> bool {
-				(self.filter)(call)
+				// Root bypasses all filters
+				matches!(self.caller, OriginCaller::system(#system_path::Origin::<#runtime>::Root))
```

Comment:
```suggestion
				match self.caller { OriginCaller::system(#system_path::Origin::<#runtime>::Root) => true, _ => false }
```

Or use an absolute path to `matches!` here.

Reply:
indeed, done in https://github.com/paritytech/substrate/pull/9948/commits/84a9ac06bb6b08a39ed622f161ecae81f9b8db2d